src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs

```diff
@@ -168,6 +168,50 @@ public IList<ITemplate> GetTemplatesAndLangpacksFromDir(IMountPoint source, out
             return templateList;
         }
 
+        // TODO: localize the diagnostic strings
+        // checks that all the template sources are under the template root, and they exist.
+        internal bool AreAllTemplatePathsValid(IRunnableProjectConfig templateConfig, RunnableProjectTemplate runnableTemplate)
+        {
+            ITemplateEngineHost host = runnableTemplate.Source.EnvironmentSettings.Host;
+
+            if (runnableTemplate.TemplateSourceRoot == null)
+            {
+                host.LogDiagnosticMessage(string.Empty, "Authoring");
+                host.LogDiagnosticMessage(string.Format("Template: {0} - Template root is outside the specified install source location.", runnableTemplate.Name), "Authoring");
+                return false;
+            }
+
+            // check if any sources get out of the mount point
+            bool allSourcesValid = true;
+            foreach (FileSourceMatchInfo source in templateConfig.Sources)
+            {
+                try
+                {
+                    IDirectory sourceRoot = runnableTemplate.TemplateSourceRoot.DirectoryInfo(source.Source);
+
+                    if (!sourceRoot.Exists)
+                    {   // non-existant directory
+                        allSourcesValid = false;
+                        host.LogDiagnosticMessage(string.Empty, "Authoring");
+                        host.LogDiagnosticMessage(string.Format("Template: '{0}'", runnableTemplate.Name), "Authoring");
+                        host.LogDiagnosticMessage(string.Format("\tTemplateSourceRoot = '{0}'", runnableTemplate.TemplateSourceRoot.FullPath), "Authoring");
+                        host.LogDiagnosticMessage(string.Format("\tSource '{0}' in template does not exist.", source.Source), "Authoring");
+                        host.LogDiagnosticMessage(string.Format("\tSource path relative to install location: '{0}'", sourceRoot.FullPath), "Authoring");
+                    }
+                }
+                catch
+                {   // outside the mount point root
```

Comment:
We should leave a note to revisit this if we fix the NullRef that happens looking up DirectoryInfo when it escapes the root